packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin+AppDelegate.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import Flutter
+
+private let completionHandlerIdKey = "completionHandlerId"
+
+extension AmplifyPushNotificationsPlugin {
+    public func application(
+        _ application: UIApplication,
+        didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data
+    ) {
+        let token = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
+        sharedEventsStreamHandlers.tokenReceived.sendEvent(payload: ["token": token])
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [AnyHashable : Any] = [:]
+    ) -> Bool {
+        if let remoteNotification = launchOptions[UIApplication.LaunchOptionsKey.remoteNotification] as? [AnyHashable: Any] {
+            // If the App is awaken by a remote notification when remote notifications background mode is enabled
+            if application.applicationState == .background {
+                awakenInBackground = true
+            }
+
+            // When the App is launched (from terminated state) by tapping on a notification
+            // This happens when remote notifications background mode is NOT enabled
+            if application.applicationState != .background {
+                launchNotification = remoteNotification
+            }
+        }
+
+        return true
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didFailToRegisterForRemoteNotificationsWithError error: Error
+    ) {
+        sharedEventsStreamHandlers.tokenReceived.sendError(
+            error: FlutterError(code: "DeviceTokenError", message: error.localizedDescription, details: error)
+        )
+    }
+
+    public func application(
+        _ application: UIApplication,
+        didReceiveRemoteNotification userInfo: [AnyHashable : Any],
+        fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void
+    ) -> Bool {
+        if UIApplication.shared.applicationState == .background {
```

Comment:
Can you switch on `applicationState` instead of if/else?

Reply:
Yea good to use `@unknown default` to avoid handling of unexpected case that falls into the `else`.